### ISSUE TITLE
Gigabit internet makes Steam incredibly unstable

### ISSUE THREAD

**Issue by mineLdiver** [ValveSoftware/steam-for-linux#10815]
#### Your system information

* Steam client version (build number or date): 1709846872
* Distribution (e.g. Ubuntu): Arch Linux
* Opted into Steam client beta?: No
* Have you checked for system updates?: Yes
* Steam Logs: [steam-logs.tar.gz](https://github.com/ValveSoftware/steam-for-linux/files/15082247/steam-logs.tar.gz)
* GPU: Nvidia 1660 SUPER

#### Please describe your issue in as much detail as possible:
For whatever reason, Steam has severe issues with 1000 Mb/s internet connection on Arch Linux.
It can take several or tens of minutes to start, and even then it might just crash during that:
![image](https://github.com/ValveSoftware/steam-for-linux/assets/19170211/2f7b2af5-9970-44f9-8bce-467e2aebc225)

However, the issues don't end here. If it does successfully start, it can randomly snap into offline mode, preventing any downloads.
And if it does download, it sits at around 40 Mb/s, despite being on a 1000 Mb/s connection. Very, very rarely it goes to about 500 Mb/s.

I know that this is somehow related to the 1000 Mb/s connection because tethering a phone and then starting Steam on mobile data fixes all the issues.

What we have tried:
1. Both native and flatpak versions of Steam.
2. Disabling HTTP2 as per https://github.com/ValveSoftware/steam-for-linux/issues/10248 but it didn't seem to have a major impact on anything.
3. Checking if NetworkManager is enabled and running. It is.

#### Steps for reproducing this issue:

1. Having a 1000 Mb/s internet connection.
2. Opening Steam

**Comment by soleofthesea** [comment]
Sounds like an issue with the Ethernet driver then. Do you actually get 1gbit download speed (like on fast.com or something)? Also try manually setting the connection from full duplex to half duplex or vice versa?

**Comment by mineLdiver** [comment]
Yes, everywhere else except Steam the download speed seems to be consistently receiving the gigabit treatment:
![image](https://github.com/ValveSoftware/steam-for-linux/assets/19170211/feea5265-53f5-4c57-9930-747e56969090)

Steam also rarely goes up to 600 Mb/s, but, again, rarely, compared to everything else.
![image](https://github.com/ValveSoftware/steam-for-linux/assets/19170211/0cc62452-dbe6-41f8-85c1-640a73a7a97d)

**Comment by soleofthesea** [comment]
Again, try setting your Ethernet to 100Mbit and see if the issue persists.

Also, how long ago did this start / has it ever worked under Arch? If it's recent it could be because your ISP is trying to block Steam, just sayin'.
